python/cuspatial/cuspatial/tests/test_geocolumn_accessor.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2022 NVIDIA Corporation
+
+import cupy as cp
+import geopandas as gpd
+import pytest
+from shapely.geometry import (
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Polygon,
+)
+
+import cuspatial
+
+
+@pytest.mark.parametrize(
+    "range, expected",
+    [[slice(0, 3), [0, 3, 4, 5]], [slice(3, 6), [0, 30, 40, 41]]],
+)
+def test_GeoColumnAccessor_polygon_offset(range, expected):
+    gpdf = cuspatial.from_geopandas(
+        gpd.read_file(gpd.datasets.get_path("naturalearth_lowres"))
+    )
+    shorter = gpdf[range]["geometry"]
+    expected = cp.array(expected)
+    got = shorter.polygons.geometry_offset
+    assert cp.array_equal(got, expected)
+
+
+@pytest.mark.parametrize(
+    "range, expected",
+    [
+        [[0, 2], [0, 2, 5]],
```

Comment:
I think what's tested here is gather indices but not a slice range.. IMO a range should be contiguous.

Reply:
Yes I'm testing both, a range and a spread out gather. Users will expect to be able to gather their `GeoSeries` in an arbitrary fashion.